### PR TITLE
clean up interfaces

### PR DESCRIPTION
**Pull request by KochTobi**
- [x] remove currentLocation from interfaces and use case
- [x] remove clearSelection method from controller
- [x] separate availableLocations into its own interface

### CODE REVIEW

**Comment by KochTobi** [comment]
This should now be ready.

File: src/main/groovy/life/qbic/portal/sampletracking/DependencyManager.groovy [hunk 1]
```diff
@@ -103,17 +104,26 @@ class DependencyManager {
...
+            controlElementsPresenter = new ControlElementsPresenter(this.viewModel)
+        } catch (NullPointerException e){
+            log.error("Could not setup presenters. NullPointer detected.", e)
+        } catch (Exception e) {
+            log.error("Unexpected exception during presenter setup.", e)
+        }
         try {
             SampleTrackingQueryDataSource trackingInfoCenter = SampleTracker.createSampleTrackingInformation(trackingServices.get(0), this.serviceUser)
-            this.queryInfoInteractor = new QuerySampleTrackingInfo(trackingInfoCenter, this.viewModel as SampleTrackingQueryOutput)
+            this.queryInfoInteractor = new QuerySampleTrackingInfo(trackingInfoCenter, sampleListPresenter, controlElementsPresenter)
```
Comment (sven1103) [inline src/main/groovy/life/qbic/portal/sampletracking/DependencyManager.groovy]: two Presenters? Why? I suggest two make the two use cases explicit

**Review by sven1103 — CHANGES_REQUESTED** [review]

File: src/main/groovy/life/qbic/portal/sampletracking/DependencyManager.groovy (hunk 1 above)
Comment (KochTobi) [reply src/main/groovy/life/qbic/portal/sampletracking/DependencyManager.groovy]: So completely separate the use cases with interactor and all the rest?

Comment (KochTobi) [reply src/main/groovy/life/qbic/portal/sampletracking/DependencyManager.groovy]: The diagram shows why two.

**Review by sven1103 — APPROVED** [review]

**Pull request merged**